### DISCUSSION TITLE
sessionStorage read fails when storage partition is enabled

### DISCUSSION THREAD

**Discussion by vinothvk** [privacysandbox/privacy-sandbox-dev-support#89] [Replacing cross-site tracking] [answered]
Hello

I am using Chrome 109 with flag 'Experimental third-party storage partitioning' enabled and 'Block all third party cookies selected'
My top level site green.com embeds a site red.com using iframe. Site red.com trying to set the sessionStorage using
sessionStorage.setItem but fails with error
'DOMException: Failed to read the 'sessionStorage' property from 'Window': Access is denied for this document'
Is this a bug ?

**Comment by miketaylr** [comment]
Hi @vinothvk - this is currently expected behavior. When a user enables "Block all third party cookies", it also disables all storage in a 3rd party context. Once partitioned 3P storage is launched, it may be the case that we later enable [partitioned 3P storage] for the "Block 3P Cookies" settings, but have no plans to announce today.

(edited for clarity, hopefully)

**Reply by vinothvk** [reply]
@miketaylr  thanks for the immediate response
So when the storage partitioning feature comes out of trial, red.com in my example should able to read and write partitioned session storage even though 'Block 3P Cookies' settings enabled or 'third party cookies are blocked'. Is that correct ?

**Reply by vinothvk** [reply]
@miketaylr can you please confirm the above ?

**Reply by miketaylr** [reply]
Apologies for the delay. 

> So when the storage partitioning feature comes out of trial, red.com in my example should able to read and write partitioned session storage even though 'Block 3P Cookies' settings enabled or 'third party cookies are blocked'. Is that correct ?

No, that's not correct. That may come at some future point, but not when we first ship partitioned storage.

**Reply by vinothvk** [reply]
@miketaylr lot of embedded applications scenarios out there today uses web storage. If privacy sandbox becomes live, without the partitioned storage supporting this use case, it will be problematic for all those embedded trusted third party domain applications to function. We need a migration path for this scenario or guidance ?
For cookies we have CHIPS but when it comes to webstorage we lack the solution.

**Reply by miketaylr** [reply]
@vinothvk perhaps I don't follow - once storage partitioning ships, embedded applications will still have storage, it will just be partitioned with the top-level site. Or do you need access to _unpartitioned_ storage set from the 1P context?

**Reply by vinothvk** [reply]
@miketaylr  The use case is, 3rd party embedded applications to have access to their own partitioned storage, when privacy sandbox rolls out live / chrome's block 3rd party cookies settings enabled.

**Reply by vinothvk** [reply]
@miketaylr any updates on this

**Reply by miketaylr** [reply, marked as answer]
Apologies for the delay.

> 3rd party embedded applications to have access to their own partitioned storage, when privacy sandbox rolls out live

This will be true, once we ship partitioned storage

> chrome's block 3rd party cookies settings enabled.

Initially, when 3rd party cookie blocking is turned on, storage will be disabled (this is the current state today - no change from the status quo). We haven't yet scoped out the work to turn on partitioned storage when 3rd party cookie blocking setting is enabled, but it's on our list of priorities for this year. 

Does that answer your question @vinothvk?

**Reply by vinothvk** [reply]
Thanks for confirming @miketaylr . Appreciate it.

I have a basic question not related to this thread, I believe privacy sandbox and block 3p cookies in chrome are two different things. Is that correct ?

If so, what are the differences? 
And what is the correct way to enable and test just the  privacy sandbox in chrome ?

**Reply by miketaylr** [reply]
Good question. "Privacy Sandbox" is an effort to prevent cross-site tracking and improve privacy generally. See https://privacysandbox.com/open-web/#how-works-on-web-hero. Removing 3P cookies is part of that plan, but it encompasses many other things (new APIs related to Ads or Ads use cases, reducing passive entropy collection, etc).

**Reply by vinothvk** [reply]
@miketaylr  in march 2023 you have mentioned below, by any chance is it planned to be released this year ?

---------------------------------------------
Initially, when 3rd party cookie blocking is turned on, storage will be disabled (this is the current state today - no change from the status quo). We haven't yet scoped out the work to turn on partitioned storage when 3rd party cookie blocking setting is enabled, but it's on our list of priorities for this year.

**Reply by miketaylr** [reply]
Hey @vinothvk - yep, this has landed just this week. You should be able to test in the latest Canary version (119.0.6039.0 or higher).

https://chromiumdash.appspot.com/commit/6bf7d85273b2192b3255e5e7cd345a3b69dd4d4b

**Reply by vinothvk** [reply]
@miketaylr  Just tested in that version and it is working as expected. Thanks to you and the chrome team for getting this into chrome and enabling site embedding use cases to work

**Reply by miketaylr** [reply]
Thanks for verifying!

**Reply by vinothvk** [reply]
@miketaylr  one more question, does microsoft edge, which uses chromium, will reflect the above change ?

**Reply by miketaylr** [reply]
I don't have any insight into Microsoft's plans around 3PCD or Storage Partitioning, unfortunately. I could speculate, but probably not useful. :)

**Reply by vinothvk** [reply]
understood. by any chance, do you know a forum like this for Edge where i can clarify ?
